src/content/app/tools/blast/types/blastApiResponse.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export type BlastApiResponse = {
+  result: {
+    hits: Hit[];
+  };
+};
+
+type Hit = HSP[];
```

Comment:
What I am seeing in an example response from the BLAST api is:

```
{
  "result": {
    "program": "blastn",
    ...
    "hits": [
      {
        "hit_num": 1,
        "hit_def": "EG:4 dna:chromosome chromosome:GRCh37:4:1:191154276:1 REF",
        "hit_db": "EG",
        "hit_id": "4",
        "hit_acc": "4",
        "hit_desc": "dna:chromosome chromosome:GRCh37:4:1:191154276:1 REF",
        ...
        "hit_hsps": [
            {
              "hsp_num": 1,
              "hsp_score": 260,
              ...
            },
        ]
      }
    ]
  }
}
```

In other words, a hit is not an array of HSPs, but an object that contains a `hit_hsps` field that is an array of HSPs. How is the above-defined type working for you?

Reply:
I have noticed some problem with the blastApiSlice endpoint. It is bringing back all data from the response, not matching with the type specified. Is it supposed to be like that?

Reply:
Updated, though we will need to confirm with Steve on usage of HSP.